spring-cloud-huawei-service-engine/service-engine-discovery/src/main/java/com/huaweicloud/authentication/ServicecombInstanceService.java

```diff
@@ -0,0 +1,131 @@
+ */
+
+package com.huaweicloud.authentication;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.util.concurrent.UncheckedExecutionException;
+import com.huaweicloud.governance.authentication.Const;
+import com.huaweicloud.governance.authentication.MicroserviceInstanceService;
+import com.huaweicloud.servicecomb.discovery.registry.ServiceCombRegistration;
+import org.apache.servicecomb.service.center.client.ServiceCenterClient;
+import org.apache.servicecomb.service.center.client.model.Microservice;
+import org.apache.servicecomb.service.center.client.model.MicroserviceInstance;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.beans.PropertyDescriptor;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+public class ServicecombInstanceService implements MicroserviceInstanceService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ServicecombInstanceService.class);
+
+    private final ServiceCombRegistration registration;
+
+    private final ServiceCenterClient client;
+
+    public ServicecombInstanceService(ServiceCombRegistration registration, ServiceCenterClient client) {
+        this.registration = registration;
+        this.client = client;
+    }
+
+    @Override
+    public void setPublickey(String publicKeyEncoded) {
+        registration.getMicroserviceInstance().getProperties().put(Const.INSTANCE_PUBKEY_PRO,
+                publicKeyEncoded);
+    }
+
+    @Override
+    public String getInstanceId() {
+        return registration.getMicroserviceInstance().getInstanceId();
+    }
+
+    @Override
+    public String getServiceId() {
+        return registration.getMicroservice().getServiceId();
+    }
+
+    @Override
+    public String getPublicKeyFromInstance(String instanceId, String serviceId) {
+        MicroserviceInstance instances = getOrCreate(serviceId, instanceId);
+        if (instances != null) {
+            return instances.getProperties().get(Const.INSTANCE_PUBKEY_PRO);
+        } else {
+            LOGGER.error("not instance found {}-{}, maybe attack", instanceId, serviceId);
+            return "";
+        }
+    }
+
+    @Override
+    public Map<String, String> getProperties(String serviceId, String instanceId, String propertyName) {
+        Microservice microservice = getMicroservice(serviceId);
+        Map<String, String> property = microservice.getProperties();
+        try {
+            Object fieldValue = new PropertyDescriptor(propertyName, Microservice.class).getReadMethod().invoke(microservice);
+            if (fieldValue.getClass().getName().equals(String.class.getName())) {
+                property.put(propertyName, (String) fieldValue);
+            }
+        } catch (Exception e) {
+            LOGGER.warn("can't find property name: {} in microservice field.", propertyName);
+        }
+        return property;
```

Comment:
return a new map without change the microservice properties.

Reply:
fixed